tests/phpunit/tests/post/getPageByTitle.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+
+/**
+ * @group post
+ *
+ * @covers ::get_page_by_title
+ */
+class Tests_Post_GetPageByTitle extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+	}
+	/**
+	 * @ticket 36905
+	 */
+	public function test_get_page_by_title() {
+		$page_id = self::factory()->post->create_object(
+			array(
+				'post_title' => 'Some Page',
+				'post_type'  => 'page',
+			)
+		);
+		$page    = get_page_by_title( 'some Page' );
+		$this->assertEquals( $page_id, $page->ID );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_get_page_by_title_second_call_uses_cache() {
+		global $wpdb;
+
+		$page_id = self::factory()->post->create_object(
+			array(
+				'post_title' => 'Some Page',
+				'post_type'  => 'page',
+			)
+		);
+
+		$page = get_page_by_title( 'some Page' );
+		$this->assertEquals( $page_id, $page->ID );
+
+		$num_queries = $wpdb->num_queries;
+
+		$page = get_page_by_title( 'some Page' );
+		$this->assertEquals( $page_id, $page->ID );
+
+		$this->assertSame( $num_queries + 1, $wpdb->num_queries );
```

Comment:
I think the `+1` is in error. The query is cached on line 40, so the number of queries shouldn't change from that on line 43.

BTW, you can use `get_num_queries()` instead of messing around with the `$wpdb` global if you wish.

:1234: Same appears to apply in the other tests too.